presto-main/src/test/java/com/facebook/presto/metadata/TestDiscoveryNodeManager.java

```diff
@@ -71,19 +71,19 @@ public void setup()
         testHttpClient = new TestingHttpClient(input -> new TestingResponse(OK, ArrayListMultimap.create(), ACTIVE.name().getBytes()));
 
         expectedVersion = new NodeVersion("1");
-        coordinator = new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.2.8"), expectedVersion, true);
-        resourceManager = new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.2.9"), expectedVersion, false, true);
-        currentNode = new InternalNode(nodeInfo.getNodeId(), URI.create("http://192.0.1.1"), expectedVersion, false);
+        coordinator = new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.2.8"), expectedVersion, true, "testPool1");
+        resourceManager = new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.2.9"), expectedVersion, false, true, "testPool1");
+        currentNode = new InternalNode(nodeInfo.getNodeId(), URI.create("http://192.0.1.1"), expectedVersion, false, "testPool2");
 
         activeNodes = ImmutableSet.of(
                 currentNode,
-                new InternalNode(UUID.randomUUID().toString(), URI.create("http://192.0.2.1:8080"), expectedVersion, false),
-                new InternalNode(UUID.randomUUID().toString(), URI.create("http://192.0.2.3"), expectedVersion, false),
+                new InternalNode(UUID.randomUUID().toString(), URI.create("http://192.0.2.1:8080"), expectedVersion, false, "testPool3"),
+                new InternalNode(UUID.randomUUID().toString(), URI.create("http://192.0.2.3"), expectedVersion, false, "testPool3"),
                 coordinator,
                 resourceManager);
         inactiveNodes = ImmutableSet.of(
-                new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.3.9"), NodeVersion.UNKNOWN, false),
-                new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.4.9"), new NodeVersion("2"), false));
+                new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.3.9"), NodeVersion.UNKNOWN, false, "testPool4"),
+                new InternalNode(UUID.randomUUID().toString(), URI.create("https://192.0.4.9"), new NodeVersion("2"), false, "testPool4"));
```

Comment:
I am a bit confused by this test case.
Is the pool information used anywhere in discovery ?
for e.g. if I move currentNode from testPool2 to say testPool3/4/5 would the test case still pass?

Reply:
The pool information is used from selector inside DiscoveryNodeManager to create internal node. However it looks like this test is not asserting equality on content of InternalNode.  So even if we change any properties of the node (except node identifier) while announcing using selector, the test would work fine. I will fix the test, so that it looks at all other fields not just the node identifier.